File: genfile.py
import shutil
import os

def copy_lua_to_src():
    source_dir = 'lua'
    target_dir = 'src'

    # Check if the source directory exists to avoid errors
    if not os.path.exists(source_dir):
        print(f"Error: The source directory '{source_dir}' does not exist.")
        return

    # Create the target directory if it doesn't already exist
    if not os.path.exists(target_dir):
        os.makedirs(target_dir)
        print(f"Created target directory: {target_dir}")

    # Iterate through all files in the source directory
    for filename in os.listdir(source_dir):
        source_path = os.path.join(source_dir, filename)
        target_path = os.path.join(target_dir, filename)

        # Ensure we are only copying files (not subfolders)
        if os.path.isfile(source_path):
            shutil.copy2(source_path, target_path)
            print(f"Copied: {filename}")

    print("\nFile transfer complete.")

if __name__ == "__main__":
    copy_lua_to_src()

File: genfile.lua
local source_dir = "lua"
local target_dir = "src"

-- 1. Create the target directory if it doesn't exist
-- 'mkdir -p' works on Unix; 'if not exist' syntax is for Windows
local function create_dir(path)
    if package.config:sub(1,1) == "\\" then -- Windows check
        os.execute("if not exist " .. path .. " mkdir " .. path)
    else
        os.execute("mkdir -p " .. path)
    end
end

-- 2. Copy the files
local function copy_files(src, dest)
    create_dir(dest)
    
    local cmd
    if package.config:sub(1,1) == "\\" then
        -- Windows command (xcopy /y copies files, /e would do folders)
        cmd = string.format("xcopy %s\\* %s\\ /y /i", src, dest)
    else
        -- Unix/Linux/macOS command
        cmd = string.format("cp -r %s/* %s/", src, dest)
    end
    
    local success = os.execute(cmd)
    if success then
        print("Successfully copied files from " .. src .. " to " .. dest)
    else
        print("Error: Failed to copy files.")
    end
end

copy_files(source_dir, target_dir)